go-controller/pkg/ovn/policy.go

```diff
@@ -1492,3 +1495,80 @@ func (oc *Controller) shutdownHandlers(np *networkPolicy) {
 		oc.watchFactory.RemoveServiceHandler(handler)
 	}
 }
+
+// iterateRetryNetworkPolicies checks if any outstanding NetworkPolicies exist
+// then tries to re-add them if so
+// updateAll forces all policies to be attempted to be retried regardless
+func (oc *Controller) iterateRetryNetworkPolicies(updateAll bool) {
+	oc.retryNetPolices.retryMutex.Lock()
+	defer oc.retryNetPolices.retryMutex.Unlock()
+	now := time.Now()
+	for namespacedName, npEntry := range oc.retryNetPolices.entries {
+		if npEntry.ignore {
+			continue
+		}
+		// check if we need to create
+		var policyToCreate *knet.NetworkPolicy
+		if npEntry.newObj != nil {
+			p := npEntry.newObj.(*knet.NetworkPolicy)
+			// get the latest version of the new policy from the informer, if it doesn't exist we are not going to
+			// create the new policy
+			np, err := oc.watchFactory.GetNetworkPolicy(p.Namespace, p.Name)
+			if err != nil && kerrors.IsNotFound(err) {
+				klog.Infof("%s policy not found in the informers cache, not going to retry policy create", namespacedName)
+				npEntry.newObj = nil
+			} else {
+				policyToCreate = np
```

Comment:
NIT: Is there any value in updating `npEntry.newObj = np` so we have the latest and greatest copy of it?

Reply:
not sure about this one I will defer to @trozet  since this is his code I just moved to policy.go as is